prboom2/CMakeLists.txt

```diff
@@ -106,8 +118,8 @@ endif()
 
 option(WITH_VORBISFILE "Use vorbisfile if available" ON)
 if(WITH_VORBISFILE)
-    find_package(Vorbis)
-    if(VORBIS_FOUND)
+    find_package(Vorbis COMPONENTS File)
```

Comment:
What does the difference here do?

Reply:
The upstream config for Vorbis declares its 3 libraries as components: `Vorbis`, `Enc`, and `File`, for respectively vorbis, vorbisenc, and vorbisfile.

Currently, it seems like all components are included by default, but there are no guarantees that it will stay the same. CMake does not provide any guarantee for the default with components, leaving it to the package to decide. Since we specifically want vorbisfile, we ask for it explicitly.

In the Find module, I reflect that requirement with:
```cmake
find_package_handle_standard_args(
  Vorbis
  REQUIRED_VARS "Vorbis_File_LIBRARY" "Vorbis_Vorbis_LIBRARY"
                "Vorbis_INCLUDE_DIR"
  HANDLE_COMPONENTS)
```
Where I declare that only `File` and `Vorbis` are required, and that it should handle components.